### PR TITLE
DAOS-2738 tests: Fix return code of launch.py

### PR DESCRIPTION
**Pull request by phender**
Launch.py should return a zero return code if it is able to run all the
tests regrdless of twhether or not those tests pass or fail to work
correctly in CI.

Signed-off-by: Phil Henderson <phillip.henderson@intel.com>

### CODE REVIEW

File: src/tests/ftest/launch.py [hunk 1]
```diff
@@ -564,7 +564,8 @@ def main():
     test_files = get_test_files(test_list)
 
     # Run all the tests
-    exit(run_tests(test_files, tag_filter, args))
+    run_tests(test_files, tag_filter, args)
+    exit(0)
```
Comment (brianjmurrell) [inline src/tests/ftest/launch.py]: As discussed on IM, we just want to mask (i.e. bitwise-or) out a return code of 1 and 8 from the `avocado` exit status.

Comment (phender) [reply src/tests/ftest/launch.py]: This has been implemented.

File: src/tests/ftest/launch.py [hunk 2]
```diff
@@ -564,7 +564,25 @@ def main():
...
+    status = run_tests(test_files, tag_filter, args)
+
+    # Process the avocado run return codes and only treat job and command
+    # failures as errors.
+    ret_code = 0
+    if status == 0:
+        print("All avocado tests passed!")
+    else:
+        if status & 1 == 1:
+            print("Detected one or more avocado test failures!")
```
Comment (ravalsam) [inline src/tests/ftest/launch.py]: Need ret_code = 1?

File: src/tests/ftest/launch.py [hunk 3]
```diff
@@ -564,7 +564,25 @@ def main():
...
+        if status & 8 == 8:
+            print("Detected one or more interrupted avocado jobs!")
```
Comment (ravalsam) [inline src/tests/ftest/launch.py]: Need ret_code = 1?

File: src/tests/ftest/launch.py (hunk 2 above)
Comment (phender) [reply src/tests/ftest/launch.py]: No, that is what this patch is fixing.  For use with ftest.sh we want launch.py to return a successful status if it is able to execute the avocado tests, regardless of whether they pass/fail/skip/cancel/etc.  We only want a bad status from launch.py if it encounters errors running the avocado commands.

File: src/tests/ftest/launch.py (hunk 3 above)
Comment (phender) [reply src/tests/ftest/launch.py]: No, that is what this patch is fixing.  For use with ftest.sh we want launch.py to return a successful status if it is able to execute the avocado tests, regardless of whether they pass/fail/skip/cancel/etc.  We only want a bad status from launch.py if it encounters errors running the avocado commands.

File: src/tests/ftest/launch.py (hunk 2 above)
Comment (ravalsam) [reply src/tests/ftest/launch.py]: ok , LGTM

**Pull request merged**